lib/mocha.js

```diff
@@ -424,22 +448,42 @@ Mocha.prototype.run = function(fn){
   var options = this.options;
   options.files = this.files;
   var runner = new exports.Runner(suite, options.delay);
-  var reporter = new this._reporter(runner, options);
+  // For each loaded reporter constructor, create
+  // an instance and initialize it with the runner
+  var reporters = this._reporters.map(function (reporterConfig) {
+    var _reporter = reporterConfig[0]
+      , path = reporterConfig[1];
+
+    return new _reporter(runner, options, path);
+  });
   runner.ignoreLeaks = false !== options.ignoreLeaks;
   runner.fullStackTrace = options.fullStackTrace;
   runner.asyncOnly = options.asyncOnly;
   if (options.grep) runner.grep(options.grep, options.invert);
   if (options.globals) runner.globals(options.globals);
-  if (options.growl) this._growl(runner, reporter);
+
+  // Use only the first reporter for growl, since we don't want to
+  // send several notifications for the same test suite
+  if (options.growl) this._growl(runner, reporters[0]);
+
   if (options.useColors !== undefined) {
     exports.reporters.Base.useColors = options.useColors;
   }
+
   exports.reporters.Base.inlineDiffs = options.useInlineDiffs;
 
   function done(failures) {
+    var remain = reporters.length;
+
+    reporters.forEach(function (reporter) {
       if (reporter.done) {
-          reporter.done(failures, fn);
-      } else fn && fn(failures);
+        reporter.done(failures, function (failures) {
+          if (--remain === 0) fn && fn(failures);
+        });
+      } else if (--remain === 0 && fn) {
+        fn(failures);
+      }
```

Comment:
I know it's not because of your code, just discussing it, but I don't see clear why the reporter could/can modify the failures object that will be passed to the `run`'s callback. I couldn't find any of our bundled reporters with a `done` method either. Any thoughts @mochajs/mocha?